huaweicloud/services/ccm/resource_huaweicloud_ccm_private_ca.go

```diff
@@ -0,0 +1,514 @@
+			int64(utils.PathSearch("create_time", getPrivateCARespBody, float64(0)).(float64))/1000, false)),
+		d.Set("expired_at", utils.FormatTimeStampRFC3339(
+			int64(utils.PathSearch("not_after", getPrivateCARespBody, float64(0)).(float64))/1000, false)),
+		d.Set("free_quota", utils.PathSearch("free_quota", getPrivateCARespBody, nil)),
+	)
+
+	if err := mErr.ErrorOrNil(); err != nil {
+		return diag.Errorf("error setting CCM private CA fields: %s", err)
+	}
+
+	return nil
+}
+
+func flattenDistinguishedName(resp interface{}) []interface{} {
+	curJson := utils.PathSearch("distinguished_name", resp, make([]interface{}, 0))
+	curArray := curJson.(map[string]interface{})
+	rst := make([]interface{}, 0, 1)
+	rst = append(rst, map[string]interface{}{
+		"common_name":         curArray["common_name"],
+		"country":             curArray["country"],
+		"state":               curArray["state"],
+		"locality":            curArray["locality"],
+		"organization":        curArray["organization"],
+		"organizational_unit": curArray["organizational_unit"],
+	})
+	return rst
+}
+
+func flattenCrlConfiguration(resp interface{}) []interface{} {
+	curJson := utils.PathSearch("crl_configuration", resp, make([]interface{}, 0))
+	curArray := curJson.(map[string]interface{})
+	rst := make([]interface{}, 0, 1)
+	rst = append(rst, map[string]interface{}{
+		"crl_name":        curArray["crl_name"],
+		"obs_bucket_name": curArray["obs_bucket_name"],
+		"valid_days":      curArray["valid_days"],
+		"crl_dis_point":   curArray["crl_dis_point"],
+	})
+	return rst
+}
+
+func resourcePrivateCADelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	var (
+		privateCAHttpUrl = "v1/private-certificate-authorities/{id}"
+		privateCAProduct = "ccm"
+	)
+	privateCAClient, err := cfg.NewServiceClient(privateCAProduct, region)
+	if err != nil {
+		return diag.Errorf("error deleting ccm Client: %s", err)
+	}
+	privateCAPath := privateCAClient.Endpoint + privateCAHttpUrl
+	privateCAPath = strings.ReplaceAll(privateCAPath, "{id}", d.Id())
+	privateCAOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+			204,
+		},
+	}
+
+	// get and check CA status, if not expired or disable then disable it.
+	getPrivateCAResp, err := privateCAClient.Request("GET", privateCAPath, &privateCAOpt)
+	if err != nil {
+		return diag.Errorf("error getting private CA: %s", err)
+	}
+	getPrivateCARespBody, err := utils.FlattenResponse(getPrivateCAResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	status := utils.PathSearch("status", getPrivateCARespBody, nil).(string)
+	if !(status == "EXPIRED" || status == "DISABLE") {
+		disablePrivateCAPath := privateCAPath + "/disable"
+		_, err = privateCAClient.Request("POST", disablePrivateCAPath, &privateCAOpt)
+		if err != nil {
+			return diag.Errorf("error disabling private CA: %s", err)
+		}
+	}
+
+	pendingDays := "7"
```

Comment:
it's not suggest to set the default value of parameters. We should define it as **Required**